### PR TITLE
Add strategy for getting org list from DB

### PR DESCRIPTION
**Pull request by kahowell**
I tested with this config, co-opting the rhsm-subscriptions database for local testing:

```
rhsm-conduit.org-sync.strategy=databaseOrgListStrategy
rhsm-conduit.org-sync.schedule=0 0 0 * * ?
rhsm-conduit.datasource.url=jdbc:postgresql://${POSTGRESQL_SERVICE_HOST:localhost}:\
  ${POSTGRESQL_SERVICE_PORT:5432}/${DATABASE_NAME:rhsm-subscriptions}
rhsm-conduit.datasource.username=${DATABASE_USER:rhsm-subscriptions}
rhsm-conduit.datasource.password=${DATABASE_PASSWORD:rhsm-subscriptions}
rhsm-conduit.datasource.driver-class-name=org.postgresql.Driver
rhsm-conduit.datasource.platform=postgresql
```

I refactored some logic out of OrgSyncJob and into TaskManager, as this made it easier to call a full sync from JMX.

I added a few convenience methods to JMX to allow updating the org list DB, but I anticipate our initial load at least will be done with an SQL script (out of scope of this commit).

### CODE REVIEW

File: src/main/java/org/candlepin/insights/jmx/RhsmConduitJmxBean.java [hunk 1]
```diff
@@ -53,4 +65,43 @@ public void syncOrg(String orgId) {
...
+        repo.deleteAll(orgList);
+    }
+
+    private List<Organization> extractOrgList(String orgs) {
+        return Arrays.stream(orgs.split("[, \n]"))
+            .map(String::trim)
+            .filter(orgId -> !orgId.isEmpty())
+            .map(Organization::new)
+            .collect(Collectors.toList());
+    }
```
Comment (mstead) [inline src/main/java/org/candlepin/insights/jmx/RhsmConduitJmxBean.java]: Would like to see a test around this code.

File: src/main/java/org/candlepin/insights/jmx/RhsmConduitJmxBean.java [hunk 2]
```diff
@@ -53,4 +65,43 @@ public void syncOrg(String orgId) {
...
+    @ManagedOperation(description = "Remove some orgs from the database sync list")
+    @ManagedOperationParameter(name = "orgs", description = "comma-separated org list (whitespace ignored)")
+    public void removeOrgsFromSyncList(String orgs) {
+        List<Organization> orgList = extractOrgList(orgs);
+
+        log.info("Removing {} orgs from DB sync list", orgList.size());
+
+        repo.deleteAll(orgList);
+    }
+
```
Comment (mstead) [inline src/main/java/org/candlepin/insights/jmx/RhsmConduitJmxBean.java]: Would like to see some repository tests around the repo methods that we are using. If anything was broken I don't think we'd know until we deployed. Yes, it is a simple model but I think we should have them.

I'd like to see tests around what happens if there's duplicates in the list. Things like that.

Comment (mstead) [inline src/main/java/org/candlepin/insights/jmx/RhsmConduitJmxBean.java]: Would it make sense to add something that allows us to check if an org is in the list?

File: src/main/java/org/candlepin/insights/jmx/RhsmConduitJmxBean.java (hunk 1 above)
Comment (kahowell) [reply src/main/java/org/candlepin/insights/jmx/RhsmConduitJmxBean.java]: Done

File: src/main/java/org/candlepin/insights/jmx/RhsmConduitJmxBean.java (hunk 2 above)
Comment (kahowell) [reply src/main/java/org/candlepin/insights/jmx/RhsmConduitJmxBean.java]: Done

Comment (kahowell) [reply src/main/java/org/candlepin/insights/jmx/RhsmConduitJmxBean.java]: Done

Comment (kahowell) [reply src/main/java/org/candlepin/insights/jmx/RhsmConduitJmxBean.java]: Done

**Pull request merged**